src/geometry/rotation_interpolation.rs

```diff
@@ -79,3 +82,67 @@ impl<T: SimdRealField> Rotation3<T> {
         q1.try_slerp(&q2, t, epsilon).map(|q| q.into())
     }
 }
+
+impl<T: RealField, const D: usize> Rotation<T, D>
+where
+    Const<D>: DimSub<U1>,
+    ArrayStorage<T, D, D>: Storage<T, Const<D>, Const<D>>,
+    DefaultAllocator: Allocator<T, Const<D>, Const<D>, Buffer = ArrayStorage<T, D, D>>
+        + Allocator<T, Const<D>>
+        + Allocator<T, Const<D>, DimDiff<Const<D>, U1>>
+        + Allocator<T, DimDiff<Const<D>, U1>>,
+{
+    ///
+    /// Computes the spherical linear interpolation between two general rotations.
+    ///
+    /// # Examples:
+    ///
+    /// ```
+    /// # use nalgebra::geometry::Rotation3;
+    ///
+    /// let q1 = Rotation3::from_euler_angles(std::f32::consts::FRAC_PI_4, 0.0, 0.0);
+    /// let q2 = Rotation3::from_euler_angles(-std::f32::consts::PI, 0.0, 0.0);
+    ///
+    /// let q = q1.slerp(&q2, 1.0 / 3.0);
+    ///
+    /// assert_eq!(q.euler_angles(), (std::f32::consts::FRAC_PI_2, 0.0, 0.0));
+    /// ```
+    ///
+    //FIXME: merging slerp for Rotation2 and Rotation3 into this raises the trait bounds
```

Comment:
That is correct. This was basically me leaving the final API choice to later; it's not great to change that without input from everyone else.

Obviously, it'd be nice to just have one `slerp()` function that gets specialized for all of them, but if changing the bounds is going to send some people into generic hell, we can also just accept it and change the names.

I don't know how often people try to use `Complex` with a SIMD scalar though? I can't imagine it's really that often, but I suppose the real issue is that this will propagate to all of the dependent trait bounds.